Core/Graphics/Source/FrameBuffer.cpp

```diff
@@ -107,16 +139,35 @@ namespace Babylon::Graphics
         m_disposed = true;
     }
 
-    void FrameBuffer::SetBgfxViewPort(bgfx::Encoder& encoder, const ViewPort& viewPort)
+    /// Returns the desired scissor rect with the Y coordinate flipped.
+    ///
+    /// In WebGL the scissor x/y coords are the lower left of the rect, but in BGFX the scissor x/y coords are the
+    /// upper left. This function returns the given WebGL scissor rect converted to a BGFX scissor rect and makes sure
+    /// y is not negative.
+    ///
+    /// Note that this function sets the y coord to 0 if the desired scissor's height is 0 because BGFX requires the
+    /// x, y, width, and height to all be zero to disable the scissor. Without this extra check this function will set
+    /// the y coord to the framebuffer's height, which will make BGFX set an invalid scissor rect (the top and bottom
+    /// of the scissor rect will have the same value), and may crash the renderer in some cases.
+    ///
+    Rect FrameBuffer::GetFlippedScissor() const
     {
-        if (m_viewId.has_value() && viewPort.Equals(m_bgfxViewPort))
+        Rect scissor = m_desiredScissor;
+        scissor.Y = m_desiredScissor.Height == 0.0f ? 0.0f : std::max(0.0f, (Height() - m_desiredScissor.Y) - m_desiredScissor.Height);
```

Comment:
We are clamping the Y. Should we also clamp the X?

Reply:
Ya, and should we decide what an invalid scissor rect is and how invalid rects should be made valid?

Reply:
What does WebGL do?

Reply:
WebGL is fairly permissive with this. It allows negative x and y values and it allow the width and height to be set larger than the framebuffer. I updated this PR to do the same.